pkg/operator/controllers/muo/muo_controller.go

```diff
@@ -20,29 +23,45 @@ import (
 
 	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
 	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers/muo/config"
 	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+	"github.com/Azure/ARO-RP/pkg/util/pullsecret"
 )
 
 const (
 	ControllerName = "ManagedUpgradeOperator"
 
-	controllerEnabled  = "rh.srep.muo.enabled"
-	controllerManaged  = "rh.srep.muo.managed"
-	controllerPullSpec = "rh.srep.muo.deploy.pullspec"
+	controllerEnabled                = "rh.srep.muo.enabled"
+	controllerManaged                = "rh.srep.muo.managed"
+	controllerPullSpec               = "rh.srep.muo.deploy.pullspec"
+	controllerAllowOCM               = "rh.srep.muo.deploy.allowOCM"
+	controllerOcmBaseURL             = "rh.srep.muo.deploy.ocmBaseUrl"
+	controllerOcmBaseURLDefaultValue = "https://api.openshift.com"
 )
 
+var pullSecretName = types.NamespacedName{Name: "pull-secret", Namespace: "openshift-config"}
+var pullSecretOCMKey = "cloud.redhat.com"
+
+type MUODeploymentConfig struct {
+	Pullspec     string
+	ConnectToOCM bool
+	OCMBaseURL   string
+}
+
 type Reconciler struct {
-	arocli   aroclient.Interface
-	deployer Deployer
+	arocli        aroclient.Interface
+	kubernetescli kubernetes.Interface
```

Comment:
we are adding in kubecli because we need to confirm the `pullSecretOCMKey` exists in the `pull-secret` secret for a successful communication to OCM. just speaking out loud here.